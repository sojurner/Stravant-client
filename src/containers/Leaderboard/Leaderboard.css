table {
  width: 70%;
  font-family: daybold;
  margin: auto; }

.recent-dash {
  font-family: daybold;
  color: #d8cbbf;
  font-size: 2.5rem;
  text-align: center; }

td {
  text-align: center;
  padding: 12px; }

th {
  text-align: center;
  font-size: 2rem;
  padding: 0.3rem 0;
  color: white;
  background-color: #9c9ca5; }

tr:nth-child(even) {
  font-size: 1.4rem;
  background-color: #a39a8c;
  color: #e4c7ae; }

tr:nth-child(odd) {
  color: #ac997e;
  font-size: 1.4rem;
  background-color: #d8cbbf; }

@media screen and (max-width: 510px) {
  table {
    width: 100%;
    font-family: daybold;
    margin-top: 2rem; }
  td {
    text-align: center;
    padding: 12px;
    font-size: 1.2rem; }
  tr {
    font-size: 0.8rem; }
  th {
    text-align: center;
    font-size: 1rem;
    padding: 0.3rem 0;
    color: white;
    background-color: #9c9ca5; }
  .recent-dash {
    margin: 1.4rem 0 0 0;
    font-size: 1.7rem; } }
