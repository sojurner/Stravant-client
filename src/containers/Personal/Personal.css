.personal-stats {
  display: flex;
  justify-content: center; }

path {
  cursor: pointer; }

.personal-stats {
  position: absolute; }

.legend {
  background-color: #f9f9f9;
  border: 1px solid #e5e5e5;
  border-radius: 12px;
  font-size: 0.8em;
  padding-top: 2rem; }

.running-avg {
  text-align: center;
  font-family: daybold;
  font-size: 2.5rem;
  padding: auto;
  color: #a49188;
  margin: 0; }

.pie-chart-wrap {
  display: flex;
  justify-content: center; }

li {
  display: flex;
  left: 0.6rem; }

.data-details {
  position: fixed;
  font-family: daybold;
  font-size: 1.5rem;
  color: #ffc1b8;
  letter-spacing: 2px;
  background: #c5696a;
  border-radius: 0.4em;
  margin: 25rem 0;
  padding: 1rem;
  opacity: 0.9; }

.legend-container-fc25d28b8dda2e2172d549eafe0de054a26a4837 {
  width: 11rem;
  margin: 8.7rem 6.3rem;
  position: fixed;
  font-family: daybold;
  font-size: 1.5rem; }

.legend {
  padding-top: 0.5rem;
  background-color: #b3aaab; }

.spinning-globe {
  width: 200px;
  height: 200px;
  border-radius: 50%;
  margin: 3rem auto;
  background: center/cover url(../../images/earth.png);
  box-shadow: inset 0 0 25px #b07979;
  animation: world 35s linear infinite;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #a49188; }

.spinning-globe:hover {
  box-shadow: inset 0 0 25px #8b4242; }

@keyframes world {
  0% {
    background-position: 0 0; }
  100% {
    background-position: 719px 0; } }

@media screen and (max-width: 510px) {
  .pie-chart-wrap {
    margin-top: 2rem; }
  .running-avg {
    margin: 2rem; }
  .data-details {
    margin: 30rem 0; } }
