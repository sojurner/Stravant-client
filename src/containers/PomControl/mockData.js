export const currentUser = {
  info: {
    accessToken: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
    userId: 34515396,
    firstName: 'Paul',
    gender: 'M',
    profilePic:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg'
  },
  totalStats: { Biked: 0, Ran: 6.472964574269732, Swam: 0 },
  weeklyStats: {}
};

export const pomInfo = {
  pomStatus: false,
  lastPom: '',
  pomHistory: {
    'Mon, Sep 10, 2018 9:12 PM': {
      mSecond: 0,
      second: 0,
      minute: 0,
      hour: 0,
      save: true,
      showHistory: true,
      description: false
    },
    'Mon, Sep 10, 2018 11:43 PM': {
      mSecond: 0,
      second: 0,
      minute: 0,
      hour: 0,
      save: false,
      showHistory: false,
      description: false
    },
    'Mon, Sep 10, 2018 11:58 PM': {
      mSecond: 0,
      second: 0,
      minute: 0,
      hour: 0,
      save: false,
      showHistory: false,
      description: false
    }
  }
};
