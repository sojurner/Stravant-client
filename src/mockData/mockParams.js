export const tokenParams = [
  'https://www.strava.com/oauth/token',
  {
    body:
      '{"client_id":28236,"client_secret":"829f12d9bdb8ce424edf87deff4551c117d3ed36","code":"r5423rffda4c234ctwtwert2ct"}',
    headers: { 'Content-Type': 'application/json' },
    method: 'POST'
  }
];

export const aggregateParams = [
  'https://www.strava.com/api/v3/athletes/28236/stats',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];

export const weeklyParams = [
  'https://www.strava.com/api/v3/athlete/activities?before=1536123319&after=1536036919',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];

export const userInfoParams = {
  token_type: 'Bearer',
  access_token: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
  athlete: {
    id: 34515396,
    username: null,
    resource_state: 2,
    firstname: 'Paul',
    lastname: 'K',
    city: null,
    state: null,
    country: null,
    sex: 'M',
    premium: false,
    summit: false,
    created_at: '2018-09-04T22:53:48Z',
    updated_at: '2018-09-07T12:20:05Z',
    badge_type_id: 0,
    profile_medium:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    profile:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    friend: null,
    follower: null,
    email: 'pykim0591@gmail.com'
  }
};

export const userStatsParams = {
  biggest_ride_distance: null,
  biggest_climb_elevation_gain: null,
  recent_ride_totals: {
    count: 0,
    distance: 0,
    moving_time: 0,
    elapsed_time: 0,
    elevation_gain: 0,
    achievement_count: 0
  },
  recent_run_totals: {
    count: 5,
    distance: 10414.50015258789,
    moving_time: 3364,
    elapsed_time: 4832,
    elevation_gain: 9.439761638641357,
    achievement_count: 11
  },
  recent_swim_totals: {
    count: 0,
    distance: 0,
    moving_time: 0,
    elapsed_time: 0,
    elevation_gain: 0,
    achievement_count: 0
  },
  ytd_ride_totals: {
    count: 0,
    distance: 0,
    moving_time: 0,
    elapsed_time: 0,
    elevation_gain: 0
  },
  ytd_run_totals: {
    count: 5,
    distance: 10415,
    moving_time: 3364,
    elapsed_time: 4832,
    elevation_gain: 10
  },
  ytd_swim_totals: {
    count: 0,
    distance: 0,
    moving_time: 0,
    elapsed_time: 0,
    elevation_gain: 0
  },
  all_ride_totals: {
    count: 0,
    distance: 0,
    moving_time: 0,
    elapsed_time: 0,
    elevation_gain: 0
  },
  all_run_totals: {
    count: 5,
    distance: 10415,
    moving_time: 3364,
    elapsed_time: 4832,
    elevation_gain: 10
  },
  all_swim_totals: {
    count: 0,
    distance: 0,
    moving_time: 0,
    elapsed_time: 0,
    elevation_gain: 0
  }
};

export const clubDataResultParam = [
  {
    id: 473825,
    resource_state: 2,
    name: 'Stravant Club',
    profile_medium:
      'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246543/1/medium.jpg',
    profile:
      'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246543/1/large.jpg',
    cover_photo:
      'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246574/1/large.jpg',
    cover_photo_small:
      'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246574/1/small.jpg',
    sport_type: 'running',
    city: 'Denver',
    state: 'Colorado',
    country: 'United States',
    private: false,
    member_count: 5,
    featured: false,
    verified: false,
    url: 'stravant'
  }
];

export const clubActivityDataParam = [
  {
    resource_state: 2,
    athlete: { resource_state: 2, firstname: 'Pamela', lastname: 'L.' },
    name: 'Morning Run',
    distance: 6871.2,
    moving_time: 2620,
    elapsed_time: 2821,
    total_elevation_gain: 17,
    type: 'Run',
    workout_type: null
  },
  {
    resource_state: 2,
    athlete: { resource_state: 2, firstname: 'Paul', lastname: 'K.' },
    name: 'Afternoon Run',
    distance: 501.6,
    moving_time: 149,
    elapsed_time: 183,
    total_elevation_gain: 0,
    type: 'Run',
    workout_type: 0
  }
];

export const weeklyDataParams = [
  [],
  [
    {
      resource_state: 2,
      athlete: { id: 34515396, resource_state: 1 },
      name: 'Afternoon Run',
      distance: 501.6,
      moving_time: 149,
      elapsed_time: 183,
      total_elevation_gain: 0,
      type: 'Run',
      workout_type: 0,
      id: 1832950216,
      external_id: '3D3F2956-3EC2-41F4-B4B8-5C0449E93948',
      upload_id: 1964982307,
      start_date: '2018-09-10T19:44:20Z',
      start_date_local: '2018-09-10T13:44:20Z',
      timezone: '(GMT-07:00) America/Denver',
      utc_offset: -21600,
      start_latlng: [39.75, -105],
      end_latlng: [39.75, -105],
      location_city: null,
      location_state: null,
      location_country: null,
      start_latitude: 39.75,
      start_longitude: -105,
      achievement_count: 1,
      kudos_count: 0,
      comment_count: 0,
      athlete_count: 1,
      photo_count: 0,
      map: {
        id: 'a1832950216',
        summary_polyline: 'k{rqFndz_SbEeB~D|Fb@q@uApA',
        resource_state: 2
      },
      trainer: false,
      commute: false,
      manual: false,
      private: false,
      visibility: 'everyone',
      flagged: false,
      gear_id: null,
      from_accepted_tag: false,
      average_speed: 3.366,
      max_speed: 9.3,
      has_heartrate: false,
      elev_high: 1587,
      elev_low: 1585.6,
      pr_count: 0,
      total_photo_count: 0,
      has_kudoed: false
    },
    {
      resource_state: 2,
      athlete: { id: 34515396, resource_state: 1 },
      name: 'Morning Run',
      distance: 2178.6,
      moving_time: 853,
      elapsed_time: 1064,
      total_elevation_gain: 0,
      type: 'Run',
      workout_type: 0,
      id: 1832102085,
      external_id: 'EE8C6C38-CD30-4D62-87BB-0CA540BA6A48',
      upload_id: 1964114698,
      start_date: '2018-09-10T12:31:39Z',
      start_date_local: '2018-09-10T06:31:39Z',
      timezone: '(GMT-07:00) America/Denver',
      utc_offset: -21600,
      start_latlng: [39.74, -104.99],
      end_latlng: [39.75, -105],
      location_city: null,
      location_state: null,
      location_country: null,
      start_latitude: 39.74,
      start_longitude: -104.99,
      achievement_count: 4,
      kudos_count: 0,
      comment_count: 0,
      athlete_count: 1,
      photo_count: 0,
      map: {
        id: 'a1832102085',
        summary_polyline: 'wypqFnfx_SwGfDiJnQe_@ph@wEmDv@aCcDdGKgA',
        resource_state: 2
      },
      trainer: false,
      commute: false,
      manual: false,
      private: false,
      visibility: 'everyone',
      flagged: false,
      gear_id: null,
      from_accepted_tag: false,
      average_speed: 2.554,
      max_speed: 9.5,
      has_heartrate: false,
      elev_high: 1601.1,
      elev_low: 1585,
      pr_count: 0,
      total_photo_count: 0,
      has_kudoed: false
    }
  ],
  [],
  [
    {
      resource_state: 2,
      athlete: { id: 34515396, resource_state: 1 },
      name: 'Morning Run',
      distance: 3950,
      moving_time: 1069,
      elapsed_time: 1077,
      total_elevation_gain: 2.1,
      type: 'Run',
      workout_type: 0,
      id: 1827412827,
      external_id: 'DE69C2FB-C9EE-4FD8-ABCB-BDA8F89B50FD',
      upload_id: 1959242580,
      start_date: '2018-09-08T13:16:10Z',
      start_date_local: '2018-09-08T07:16:10Z',
      timezone: '(GMT-07:00) America/Denver',
      utc_offset: -21600,
      start_latlng: [39.74, -105.04],
      end_latlng: [39.74, -105.04],
      location_city: null,
      location_state: null,
      location_country: null,
      start_latitude: 39.74,
      start_longitude: -105.04,
      achievement_count: 3,
      kudos_count: 0,
      comment_count: 0,
      athlete_count: 1,
      photo_count: 0,
      map: {
        id: 'a1827412827',
        summary_polyline:
          'apqqFr_b`SmH@{ElH{Ft\\wFaAeUjP~@tKbChGsAzMbBtFdXfAzGeBdHcIhAwDo@_IqEwMfDsItBk[',
        resource_state: 2
      },
      trainer: false,
      commute: false,
      manual: false,
      private: false,
      visibility: 'everyone',
      flagged: false,
      gear_id: null,
      from_accepted_tag: false,
      average_speed: 3.695,
      max_speed: 5.6,
      has_heartrate: false,
      elev_high: 1621.2,
      elev_low: 1618.4,
      pr_count: 3,
      total_photo_count: 0,
      has_kudoed: false
    }
  ],
  [
    {
      resource_state: 2,
      athlete: { id: 34515396, resource_state: 1 },
      name: 'Lunch Run',
      distance: 1562,
      moving_time: 579,
      elapsed_time: 1086,
      total_elevation_gain: 4.7,
      type: 'Run',
      workout_type: 0,
      id: 1825796851,
      external_id: '3A54AFD1-4551-40FC-BDB3-A72100B29082',
      upload_id: 1957570524,
      start_date: '2018-09-07T17:49:18Z',
      start_date_local: '2018-09-07T11:49:18Z',
      timezone: '(GMT-07:00) America/Denver',
      utc_offset: -21600,
      start_latlng: [39.75, -105],
      end_latlng: [39.75, -105],
      location_city: null,
      location_state: null,
      location_country: null,
      start_latitude: 39.75,
      start_longitude: -105,
      achievement_count: 3,
      kudos_count: 0,
      comment_count: 0,
      athlete_count: 1,
      photo_count: 0,
      map: {
        id: 'a1825796851',
        summary_polyline: 'sxrqFvez_SShBcDlB_DeGlAsCg@mDwFvAlE|ElBuD`EfFx@g@',
        resource_state: 2
      },
      trainer: false,
      commute: false,
      manual: false,
      private: false,
      visibility: 'everyone',
      flagged: false,
      gear_id: null,
      from_accepted_tag: false,
      average_speed: 2.698,
      max_speed: 6.1,
      has_heartrate: false,
      elev_high: 1586,
      elev_low: 1583,
      pr_count: 3,
      total_photo_count: 0,
      has_kudoed: false
    },
    {
      resource_state: 2,
      athlete: { id: 34515396, resource_state: 1 },
      name: 'Morning Run',
      distance: 2222.3,
      moving_time: 714,
      elapsed_time: 1422,
      total_elevation_gain: 2.6,
      type: 'Run',
      workout_type: 0,
      id: 1825117879,
      external_id: '3CAB0E99-AEBC-4B32-9FB6-96572B190C9B',
      upload_id: 1956872229,
      start_date: '2018-09-07T11:56:11Z',
      start_date_local: '2018-09-07T05:56:11Z',
      timezone: '(GMT-07:00) America/Denver',
      utc_offset: -21600,
      start_latlng: [39.74, -104.99],
      end_latlng: [39.75, -105],
      location_city: null,
      location_state: null,
      location_country: null,
      start_latitude: 39.74,
      start_longitude: -104.99,
      achievement_count: 0,
      kudos_count: 0,
      comment_count: 0,
      athlete_count: 1,
      photo_count: 0,
      map: {
        id: 'a1825117879',
        summary_polyline:
          'q}oqFx{w_SyMq@[nDcEr@}DrG}BwAi@vHkHV]dB|@lAoDhHwALMpDua@|Zi@sB{Ax@',
        resource_state: 2
      },
      trainer: false,
      commute: false,
      manual: false,
      private: false,
      visibility: 'everyone',
      flagged: false,
      gear_id: null,
      from_accepted_tag: false,
      average_speed: 3.112,
      max_speed: 6.6,
      has_heartrate: false,
      elev_high: 1607.8,
      elev_low: 1586,
      pr_count: 0,
      total_photo_count: 0,
      has_kudoed: false
    }
  ],
  [],
  []
];

export const clubActionParam = {
  clubName: 'Stravant Club',
  clubId: 473825,
  membersStats: {
    mostRecent: {
      name: 'Paul',
      distance: 1.4110627719080175,
      activity: 'Morning Run'
    },
    Paul: { totalDistance: 7.87, totalTime: 101, membersRecent: 'Morning Run' },
    Benjamin: {
      totalDistance: 1.17,
      totalTime: 20,
      membersRecent: 'Light rail trip'
    },
    Jesse: {
      totalDistance: 278.77,
      totalTime: 5297,
      membersRecent: 'Evening Run'
    },
    Pamela: {
      totalDistance: 62.88999999999999,
      totalTime: 635,
      membersRecent: 'Afternoon Run'
    }
  }
};

export const pomHistoryParam = {
  'Wed, Sep 12, 2018 11:45 AM': { second: 2, minute: 0, hour: 0 },
  'Wed, Sep 12, 2018 12:23 PM': { second: 1, minute: 0, hour: 0 }
};

export const accessTokenParam = {
  accessToken: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
  userId: 34515396,
  firstName: 'Paul',
  gender: 'M',
  profilePic:
    'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg'
};

export const weeklyActionParam = {
  Wednesday: 2270.4,
  Monday: 2680.2,
  Saturday: 3950,
  Friday: 3784.3,
  Tuesday: 0,
  Thursday: 0,
  Sunday: 0
};

export const totalStatsParam = { Biked: 0, Ran: 7.883778744561839, Swam: 0 };
