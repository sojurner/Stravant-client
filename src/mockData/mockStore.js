export const currentUser = {
  info: {
    accessToken: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
    firstName: 'Paul',
    gender: 'M',
    profilePic:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    userId: 34515396
  },
  totalStats: {},
  weeklyStats: {}
};

export const pomStatus = {
  pomStatus: true,
  lastPom: '',
  pomHistory: {
    'Tue, Sep 11, 2018 8:39 PM': { second: 19, minute: 0, hour: 0 },
    'Tue, Sep 11, 2018 8:40 PM': { second: 3, minute: 0, hour: 0 },
    'Tue, Sep 11, 2018 10:22 PM': { second: 3, minute: 0, hour: 0 }
  }
};

export const clubInfo = {
  clubActivity: {
    Benjamin: {
      membersRecent: 'Light rail trip',
      totalDistance: 1.17,
      totalTime: 20
    },
    Jesse: {
      membersRecent: 'Evening Run',
      totalDistance: 278.77,
      totalTime: 5297
    },
    Pamela: {
      membersRecent: 'Afternoon Run',
      totalDistance: 62.88999999999999,
      totalTime: 635
    },
    Paul: { membersRecent: 'Morning Run', totalDistance: 7.87, totalTime: 101 },
    mostRecent: {
      activity: 'Morning Run',
      distance: 1.4110627719080175,
      name: 'Paul'
    }
  },
  clubInfo: { clubId: 473825, clubName: 'Stravant Club' }
};

export const mockStore = {
  ...currentUser,
  ...pomStatus,
  ...clubInfo
};

export const mockCurrentUser = {
  info: {
    accessToken: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
    userId: 34515396,
    firstName: 'Paul',
    gender: 'M',
    profilePic:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg'
  },
  totalStats: {},
  weeklyStats: {}
};
