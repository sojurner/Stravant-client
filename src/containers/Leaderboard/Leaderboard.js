import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as apiCalls from '../../helpers/apiCalls/apiCalls';
import * as clubActions from '../../actions/clubAction';
import './Leaderboard.css';

export class LeaderBoard extends Component {
  componentDidMount() {
    this.getClubActivity(this.props.currentUser.info.accessToken);
  }

  getClubActivity = async token => {
    const clubActivities = await apiCalls.getUserClubs(token);
    console.log(clubActivities);
    this.props.setClubActivity(clubActivities);
  };

  render() {
    const { clubActivity } = this.props.clubs;
    const unfilteredKeys = Object.keys(clubActivity).filter(
      key => key === 'mostRecent'
    );
    const filteredKeys = Object.keys(clubActivity).filter(
      key => key !== 'mostRecent'
    );
    const sortedByDistance = filteredKeys.sort(
      (nameA, nameB) =>
        clubActivity[nameB].totalDistance - clubActivity[nameA].totalDistance
    );
    const table = sortedByDistance.map(person => {
      return (
        <tr>
          <td>{person}</td>
          <td>{clubActivity[person].membersRecent}</td>
          <td>{Math.round(clubActivity[person].totalDistance * 100) / 100}</td>
          <td>{Math.round(clubActivity[person].totalTime * 100) / 100}</td>
        </tr>
      );
    });

    const activity = unfilteredKeys.map(recent => {
      const { distance, name, activity } = clubActivity.mostRecent;
      const roundedDistance = Math.round(distance * 100) / 100;
      const recentActivity = `${name} did a ${roundedDistance}
      mile ${activity}`;
      return (
        <div>
          <h3 className="recent-dash">{recentActivity}</h3>
        </div>
      );
    });

    return (
      <div>
        {activity}
        <table width="500">
          {!Object.keys(clubActivity).length && (
            <img
              src={'https://media.giphy.com/media/26xBNwL69IM5ng3f2/source.gif'}
              height="100"
              width="100"
              className="connect-logo"
            />
          )}
          <tbody>
            <tr>
              <th>Name</th>
              <th>Recent</th>
              <th>Distance(mi)</th>
              <th>Duration(m)</th>
            </tr>
            {table}
          </tbody>
        </table>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  currentUser: state.currentUser,
  clubs: state.clubs
});

export const mapDispatchToProps = dispatch => ({
  setClubActivity: activities =>
    dispatch(clubActions.setClubActivity(activities))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(LeaderBoard);
