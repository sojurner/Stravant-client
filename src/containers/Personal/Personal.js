import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PieChart, Legend, ToolTip } from 'react-easy-chart';
import * as apiFetch from '../../helpers/apiCalls/apiCalls';
import * as userActions from '../../actions/userAction';
import './Personal.css';
import { type } from 'os';

export class Personal extends Component {
  constructor() {
    super();
    this.state = {
      showToolTip: false,
      top: '',
      left: '',
      value: '',
      key: '',
      goal: 0
    };
  }

  componentDidMount() {
    this.getUserStats();
  }

  getUserStats = async () => {
    const { setTotalStats } = this.props;
    const { accessToken, userId } = this.props.currentUser.info;
    const userStats = await apiFetch.getAggregateStats(accessToken, userId);
    setTotalStats(userStats);
  };

  mouseOverHandler = data => {
    console.log(data);
    this.setState({
      key: data.data.key,
      value: data.data.value,
      showToolTip: true
    });
  };

  mouseOutHandler = () => {
    this.setState({ showToolTip: false });
  };

  render() {
    const { totalStats } = this.props.currentUser;
    const filteredActivity = Object.keys(totalStats).map(type => {
      const destructured = totalStats[type];
      let dataArr = [];
      let configArr = [];
      Object.keys(destructured).forEach((stat, index) => {
        const filtered = {
          key: stat,
          value: Math.round(destructured[stat] * 100) / 100,
          color: `rgb(${index}04, 1${index}9, 2${index}4)`
        };

        const config = { color: `rgb(${index}04, 1${index}9, 2${index}4)` };
        dataArr.push(filtered);
        configArr.push(config);
      });
      return (
        <div>
          <h3 className="running-avg">Running Averages</h3>
          <Legend
            className="legend-key"
            data={dataArr}
            config={configArr}
            dataId={'key'}
          />
          <PieChart
            size={375}
            data={dataArr}
            innerHoleSize={130}
            mouseOverHandler={this.mouseOverHandler}
            mouseOutHandler={event => this.mouseOutHandler(event)}
            padding={10}
          />
        </div>
      );
    });

    return (
      <div className="pie-chart-wrap">
        {!totalStats['runningTotal'] && (
          <div className="spinning-globe">Credits to Nick</div>
        )}
        {filteredActivity}
        {this.state.value &&
          this.state.showToolTip && (
            <p className="data-details">
              {`${this.state.key}: ${this.state.value}`}
            </p>
          )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  currentUser: state.currentUser
});

const mapDispatchToProps = dispatch => ({
  setTotalStats: stats => dispatch(userActions.setTotalStats(stats))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Personal);
