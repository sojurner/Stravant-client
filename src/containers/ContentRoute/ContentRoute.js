import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import Weekly from '../Weekly/Weekly';
import Personal from '../Personal/Personal';
import Leaderboard from '../Leaderboard/Leaderboard';
import PomControl from '../../containers/PomControl/PomControl';

export class ContentRoute extends Component {
  render() {
    return (
      <Switch>
        <Route exact path="/" />
        <Route exact path="/personal" component={Personal} />
        <Route exact path="/weekly" component={Weekly} />
        <Route exact path="/leaderboard" component={Leaderboard} />
        <Route exact path="/pomodoro" component={PomControl} />
      </Switch>
    );
  }
}

// 'https://images.unsplash.com/38/L2NfDz5SOm7Gbf755qpw_DSCF0490.jpg?ixlib=rb-0.3.5&q=85&fm=jpg&crop=entropy&cs=srgb&ixid=eyJhcHBfaWQiOjI0MX0&s=717f905b696bd33ced31d380b1283b60'
// 'https://images.unsplash.com/photo-1519864967-22b37f31a770?ixlib=rb-0.3.5&q=85&fm=jpg&crop=entropy&cs=srgb&ixid=eyJhcHBfaWQiOjI0MX0&s=307c453c1e85edef8468f925da9e1f74'

export default ContentRoute;
