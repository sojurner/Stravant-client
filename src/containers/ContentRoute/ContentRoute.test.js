import React from 'react';
import { MemoryRouter } from 'react-router';

import ContentRoute from './ContentRoute';
import Comparison from '../Comparison/Comparison';
import { Personal } from '../Personal/Personal';
import Leaderboard from '../Leaderboard/Leaderboard';
import * as mockStore from '../../mockData/mockStore';
import { Provider } from 'redux';

describe('ContentRoute', () => {
  it('should matchsnapshot when path matches personal', () => {
    let personalWrapper = shallow(
      <ContentRoute path="/personal" component={Personal} />
    );
    expect(personalWrapper).toMatchSnapshot();
  });

  it('should MatchSnapshot when path matches leaderboard', () => {
    let leaderboardWrapper = shallow(
      <ContentRoute path="/leaderboard" component={Leaderboard} />
    );
    expect(leaderboardWrapper).toMatchSnapshot();
  });

  it('should MatchSnapshot when path matches Compare', () => {
    let compareWrapper = shallow(
      <ContentRoute path="/compare" component={Comparison} />
    );
    expect(compareWrapper).toMatchSnapshot();
  });
});
