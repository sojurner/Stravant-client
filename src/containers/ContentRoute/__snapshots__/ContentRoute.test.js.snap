// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentRoute should MatchSnapshot when path matches Compare 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/personal"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/compare"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/leaderboard"
  />
</Switch>
`;

exports[`ContentRoute should MatchSnapshot when path matches leaderboard 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/personal"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/compare"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/leaderboard"
  />
</Switch>
`;

exports[`ContentRoute should matchsnapshot when path matches personal 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/personal"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/compare"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/leaderboard"
  />
</Switch>
`;
